### ISSUE TITLE
What is the workflow to test node-gir

### ISSUE THREAD

**Issue by cedlemo** [Place1/node-gir#33]
Hi,

I wanted to test node-gir but I am not really used to npm usage. I forked the repo, `cd` in it and when I tried to install it I had:

```
npm install node-gir                                                                                                                                                              JavaScript/node-gir  master
npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "node-gir" under a package
npm ERR! also called "node-gir". Did you name your project the same
npm ERR! as the dependency you're installing?
npm ERR! 
npm ERR! For more information, see:
npm ERR!     <https://docs.npmjs.com/cli/install#limitations-of-npms-install-algorithm>

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/cedlemo/.npm/_logs/2018-02-12T11_37_24_129Z-debug.log
```

So it looks like I should run the install outside of the local clone but the problem is that I get the node-gir that you published (I guess). How can I, for example, test the local code that I modified/ created ?

**Comment by Place1** [comment]
if you want to clone the repo and test it out then you should run `npm install` without any other arguments. The error you hit is basically telling you that you can't install a package called `node-gir` from npm in another node module (a folder with a package.json aka this repository) that is called node-gir (see package.json's "name" key). This package is already published to npm so people can install it in their projects.

So here's how you'd get node-gir running on the latest version of ubuntu:
```
sudo apt-get install libgirepository1.0-dev
sudo apt-get install libgtk-3-dev
git clone https://github.com/Place1/node-gir.git
cd nod-gir
npm install
npm test
```

The above should work on ubuntu 17.10, i'm not sure if it'll work on ubuntu 16.04 as we depend on a newer version of libgirepository1.0-dev than the version shipped in 16.04's repositories. I'm not sure what system you're trying to run on :)

**Comment by cedlemo** [comment]
> if you want to clone the repo and test it out then you should run npm install without any other 

oups sorry, it seems that in fact I did not really read/understand the doc and tried a command :sweat_smile:  . You are right it works, just FYI I use ArchLinux and I have all the needed libraries.
